### ISSUE TITLE
Profiling OPLD

### ISSUE THREAD

**Issue by AngelUMICH** [AngelUMICH/Long-Time-Dynamics#78]
### Details
OPLD has seen some great improvement in speedup mainly due to force calculations and relaxation by the LAMMPS library. However, the code is not as fast as I need it to be to do simulations in a reasonable amount of time. Therefore, more speed up is required. Profiling should be done to see where improvements could be done. Issues on how improvements could be done should be created after.

### Definition of Done
- [x] Profiled OPLD code (in parallel)
- [x] Determined what areas of OPLD code should be improved
- [x] Create issues/tickets on how to resolve them

**Comment by AngelUMICH** [comment]
This is on hold until issue #79 is resolved

**Comment by AngelUMICH** [comment]
> This is on hold until issue #79 is resolved

I can still profile but with limited number of processes which is explained in the related issue board.

**Comment by AngelUMICH** [comment]
During profiling using only one processor and finding one saddle point I found that the force calculations are the most time consuming part of the dimer code. Neighbor list calculations are also included in the previous statement. Neighbor list calculations were found to be about 8 percent of the total time of the running code. I am working on including LAMMPS commands which limit the frequency of neighbor list updating which at the moment is done at every single iteration. During most dimer searches the atoms in the defects move very small distances on the order of single lattice parameters. Therefore, the neighbor lists can be updated very seldom or maybe none at all.

**Comment by AngelUMICH** [comment]
The dimer search, mainly the force calculations including the neighbor list rebuilding, is the longest part of the code. The defect identification and the system minimization take very small amount of time compared to the dimer searches. To speed up the code the force calculations need to be optimized.
